### DISCUSSION TITLE
Qusetions about the tracing in netty server

### DISCUSSION THREAD

**Discussion by Roc-00** [open-telemetry/opentelemetry-java-instrumentation#4147] [Q&A] [answered]
After I read this (https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/netty/netty-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/server) I have a few questions.
- Is the first time to get the context is through extract the httprequest?like this:
![image](https://user-images.githubusercontent.com/84842773/133716273-2a4b9bab-1ef0-4977-86ec-a0e3f07e478a.png)

- Can I do this to attach the server context?Not using the 'context.with(key, span)'.like this:
![image](https://user-images.githubusercontent.com/84842773/133716135-d995d3f1-9383-4975-abdc-1b9dbe73961e.png)

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @Roc-00,
I'm not sure I understood your questions correctly. Do you want to instrument Netty server manually? Or do you want to extend the agent's default behavior?

**Reply by Roc-00** [reply]
yes,I am trying to instrument Netty server manually.Following the example of instrumentation, I used otle Java SDK to track the buried points of netty server.And I am not sure my understanding of instrumentation is correct.

**Reply by mateuszrzeszutek** [reply]
Okay, I understand - I'll try to answer your questions.
Regarding question 1: yes, in server instrumentations you usually (well, always) extract the context from the incoming HTTP message - some HTTP client might've propagated their trace/context using HTTP headers.
Regarding question 2: if you're using our `AttributeKey<Context> SERVER_CONTEXT` then it has to be `Context` - however as I understand it that attribute is needed by the agent to make several instrumentations work together nicely (e.g. server frameworks that internally use netty) - if you're instrumenting netty server manually then you might not need to set that one.

**Reply by Roc-00** [reply]
Thank you for your answear, I had already make it work.
